lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -220,59 +239,95 @@ AppleObjCRuntimeV2 *SwiftLanguageRuntime::GetObjCRuntime() {
 }
 
 void SwiftLanguageRuntime::SetupSwiftError() {
-  Target &target(m_process->GetTarget());
-
-  if (m_SwiftNativeNSErrorISA.hasValue())
-    return;
-
-  ConstString g_SwiftNativeNSError("__SwiftNativeNSError");
+  m_SwiftNativeNSErrorISA =
+      FindSymbolForSwiftObject(m_process->GetTarget(), RuntimeKind::Swift,
+                               "__SwiftNativeNSError", eSymbolTypeObjCClass);
+  m_initialized_swift_native_error_isa = true;
+}
 
-  m_SwiftNativeNSErrorISA = FindSymbolForSwiftObject(
-      target, g_SwiftNativeNSError, eSymbolTypeObjCClass);
+llvm::Optional<lldb::addr_t> SwiftLanguageRuntime::GetSwiftNativeNSErrorISA() {
+  if (!m_initialized_swift_native_error_isa)
+    SetupSwiftError();
+  assert(m_initialized_swift_native_error_isa);
+  return m_SwiftNativeNSErrorISA;
 }
 
 void SwiftLanguageRuntime::SetupExclusivity() {
-  Target &target(m_process->GetTarget());
-
-  ConstString g_disableExclusivityChecking("_swift_disableExclusivityChecking");
-
   m_dynamic_exclusivity_flag_addr = FindSymbolForSwiftObject(
-      target, g_disableExclusivityChecking, eSymbolTypeData);
+      m_process->GetTarget(), RuntimeKind::Swift,
+      "_swift_disableExclusivityChecking", eSymbolTypeData);
+  m_initialized_dynamic_exclusivity_flag_addr = true;
 
   Log *log(GetLogIfAnyCategoriesSet(LIBLLDB_LOG_EXPRESSIONS));
-
   if (log)
     log->Printf("SwiftLanguageRuntime: _swift_disableExclusivityChecking = %lu",
                 m_dynamic_exclusivity_flag_addr ?
                 *m_dynamic_exclusivity_flag_addr : 0);
 }
 
+llvm::Optional<lldb::addr_t>
+SwiftLanguageRuntime::GetDynamicExclusivityFlagAddr() {
+  if (!m_initialized_dynamic_exclusivity_flag_addr)
+    SetupExclusivity();
+  assert(m_initialized_dynamic_exclusivity_flag_addr);
+  return m_dynamic_exclusivity_flag_addr;
+}
+
 void SwiftLanguageRuntime::SetupABIBit() {
+  if (m_initialized_swift_classIsSwiftMask)
+    return;
   Target &target(m_process->GetTarget());
-  ConstString g_objc_debug_swift_stable_abi_bit("objc_debug_swift_stable_abi_bit");
-
-  if (FindSymbolForSwiftObject(target, g_objc_debug_swift_stable_abi_bit, eSymbolTypeAny))
+  if (FindSymbolForSwiftObject(target, RuntimeKind::ObjC,
+                               "objc_debug_swift_stable_abi_bit",
+                               eSymbolTypeAny))
     _swift_classIsSwiftMask = 2;
   else
     _swift_classIsSwiftMask = 1;
+
+  m_initialized_swift_classIsSwiftMask = true;
```

Comment:
I really think it would be cleaner to distinguish the cases where libobjc was found without the symbol from the ones were libobjc wasn't found. The latter should never happen, it means we're asking questions about the runtime before it being loaded.

Reply:
What other than logging an error would you suggest we do in that case?

Reply:
At least in debug builds, this should crash as it's an internal inconsistency. I don't care about logging that much, I'd prefer that we just make it impossible to ask the any question requiring the runtime before the runtime is setup.